action.yml

```diff
@@ -16,6 +16,13 @@ inputs:
       deploy-pr/<environment> branch and open a PR for review
     required: false
     default: .*(staging|production).*
+  include-paths:
+    description: |
+      Copy specified paths to branch.
+      Input format: JSON list of objects string with fields: if, source and target.
+      Example: [{"if": true, "source": "path/to/source1", "target": "path/to/target1"}]
```

Comment:
* is this is a folder or a file?
* what is the `if` for?
* may it happen that source and target path are different?
* is this a relative path? relative to what?

can we do something like
```
 include-paths:
  - source: local/path
    target: target/folder1
  - source: local/path2
    target: target/folder2
```

can we even copy files to the root target folder?

Reply:
_is this is a folder or a file?_
can be folder or a file

_what is the if for?_
if is used to add some condition if needed, for example if we want to copy files for wf-budget service but do not want for others

_may it happen that source and target path are different?_
yes, it's going to be different

_is this a relative path? relative to what?_
path should be relative to repository root

_can we do something like_
no, github action input do not support objects, we need to use JSON

_can we even copy files to the root target folder?_
it should be root of the repo, so should not be any problem

Reply:
_no, github action input do not support objects, we need to use JSON_
you mean Github Actions is yaml based and does not understand yaml lists as parameters?
what about https://stackoverflow.com/questions/65345432/github-actions-pass-list-of-variables-to-shell-script ?


_if is used to add some condition if needed, for example if we want to copy files for wf-budget service but do not want for others_
oh, right : we would need to add it here https://github.com/OpenGov/k8s-deployment-bnp/blob/main/.github/workflows/render-manifests.yaml#L83 and it would need a per project config so you could do something like 
```
include-paths: [{"if": matrix.env == "wf-budget", "source": "path/to/source1", "target": "path/to/target1"}]
```
like that?